lib/pos_mempool.go

```diff
@@ -54,18 +80,24 @@ type PosMempool struct {
 	// txnRegister is the in-memory data structure keeping track of the transactions in the mempool. The TransactionRegister
 	// is responsible for ordering transactions by the Fee-Time algorithm.
 	txnRegister *TransactionRegister
-	// ledger is a simple in-memory data structure that keeps track of cumulative transaction fees in the mempool.
-	// The ledger keeps track of how much each user would have spent in fees across all their transactions in the mempool.
-	ledger *BalanceLedger
 	// persister is responsible for interfacing with the database. The persister backs up mempool transactions so not to
 	// lose them when node reboots. The persister also retrieves transactions from the database when the node starts up.
 	// The persister runs on its dedicated thread and events are used to notify the persister thread whenever
 	// transactions are added/removed from the mempool. The persister thread then updates the database accordingly.
 	persister *MempoolPersister
+	// ledger is a simple data structure that keeps track of cumulative transaction fees in the mempool.
+	// The ledger keeps track of how much each user would have spent in fees across all their transactions in the mempool.
+	ledger *BalanceLedger
+	// nonceTracker is responsible for keeping track of a (public key, nonce) -> Txn index. The index is useful in
+	// facilitating a "replace by higher fee" feature. This feature gives users the ability to replace their existing
+	// mempool transaction with a new transaction having the same nonce but higher fee.
+	nonceTracker *NonceTracker
 
 	// latestBlockView is used to check if a transaction is valid before being added to the mempool. The latestBlockView
 	// checks if the transaction has a valid signature and if the transaction's sender has enough funds to cover the fee.
 	// The latestBlockView should be updated whenever a new block is added to the blockchain via UpdateLatestBlock.
+	// PosMempool only needs read-access to the block view. It isn't necessary to copy the block view before passing it
+	// to the mempool.
 	latestBlockView *UtxoView
```

Comment:
If it's an expectation that this will be a read-only field I recommend putting readonly in the name so nobody screws it up in the future.